src/contrib/persistence/Akka.Persistence.Sqlite.Tests/Legacy/SqliteLegacyJournalSpec.cs

```diff
@@ -0,0 +1,127 @@
+﻿// -----------------------------------------------------------------------
+//  <copyright file="SqliteLegacyJournalSpec.cs" company="Akka.NET Project">
+//      Copyright (C) 2009-2023 Lightbend Inc. <http://www.lightbend.com>
+//      Copyright (C) 2013-2023 .NET Foundation <https://github.com/akkadotnet/akka.net>
+//  </copyright>
+// -----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Akka.Actor;
+using Akka.Configuration;
+using Akka.TestKit;
+using FluentAssertions;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Akka.Persistence.Sqlite.Tests.Legacy
+{
+    public class SqliteLegacyJournalSpec: Akka.TestKit.Xunit2.TestKit
+    {
+        private Dictionary<string, IActorRef> _actors = new Dictionary<string, IActorRef>();
+        private readonly TestProbe _probe;
+
+        public SqliteLegacyJournalSpec(ITestOutputHelper output)
+            : base(CreateSpecConfig("Filename=file:TestDb.db"), nameof(SqliteLegacyJournalSpec), output)
+        {
+            SqlitePersistence.Get(Sys);
+            _probe = CreateTestProbe();
+        }
+        
+        private static Config CreateSpecConfig(string connectionString)
+        {
+            if(File.Exists("./TestDb.db"))
+                File.Delete("./TestDb.db");
+            File.Copy("./data/Sqlite.v1.3.0.db", "./TestDb.db");
+            
+            return ConfigurationFactory.ParseString($@"
+akka.persistence {{
+    publish-plugin-commands = on
+    journal {{
+        plugin = akka.persistence.journal.sqlite
+        sqlite {{
+            auto-initialize = on
+            connection-string = ""{connectionString}""
+        }}
+    }}
+    snapshot-store {{
+        plugin = akka.persistence.snapshot-store.sqlite
+        sqlite {{
+            auto-initialize = on
+            connection-string = ""{connectionString}""
+        }}
+    }}
+}}").WithFallback(SqlitePersistence.DefaultConfiguration());
+        }
+
+        [Fact(DisplayName = "Legacy v1.3.0 and below data (null serializer_id) should be read correctly")]
+        public void ReadLegacyDataTest()
```

Comment:
The purpose of this test is to make sure that data saved from v1.3.0 can still be recovered successfully. The Sqlite file were generated using v.1.3.0 code and is being recovered in this unit test.